### PR TITLE
instr(buffer): Fewer metrics

### PR DESCRIPTION
**Pull request by jjbayer**
Fixes: https://github.com/getsentry/team-ingest/issues/601

#skip-changelog

### CODE REVIEW

File: relay-server/src/services/buffer/envelope_buffer/mod.rs [hunk 1]
```diff
@@ -97,10 +97,6 @@ impl PolymorphicEnvelopeBuffer {
                 }?;
             }
         );
-        relay_statsd::metric!(
-            counter(RelayCounters::BufferEnvelopesWritten) += 1,
```
Comment (jjbayer) [inline relay-server/src/services/buffer/envelope_buffer/mod.rs:101]: This is already covered by the `BufferPush` metric above.

File: relay-server/src/services/buffer/envelope_buffer/mod.rs [hunk 2]
```diff
@@ -130,10 +126,6 @@ impl PolymorphicEnvelopeBuffer {
                 }?
             }
         );
-        relay_statsd::metric!(
-            counter(RelayCounters::BufferEnvelopesRead) += 1,
```
Comment (jjbayer) [inline relay-server/src/services/buffer/envelope_buffer/mod.rs:134]: This is already covered by the `BufferPop` metric above.

File: relay-server/src/services/buffer/envelope_buffer/mod.rs [hunk 3]
```diff
@@ -377,10 +369,6 @@ where
 
         match last_received_at {
             None => {
-                relay_statsd::metric!(
-                    counter(RelayCounters::BufferEnvelopeStacksPopped) += 1,
```
Comment (jjbayer) [inline relay-server/src/services/buffer/envelope_buffer/mod.rs:381]: We have a distribution metric for the number of envelopes stacks, from which we can derive this one.

File: relay-server/src/services/buffer/mod.rs [hunk 4]
```diff
@@ -243,42 +243,17 @@ impl EnvelopeBufferService {
     /// Wait for the configured amount of time and make sure the project cache is ready to receive.
     async fn ready_to_pop(
         &mut self,
-        partition_tag: &str,
         buffer: &PolymorphicEnvelopeBuffer,
         dequeue: bool,
     ) -> Option<Permit<legacy::DequeuedEnvelope>> {
-        relay_statsd::metric!(
-            counter(RelayCounters::BufferReadyToPop) += 1,
```
Comment (jjbayer) [inline relay-server/src/services/buffer/mod.rs:251]: We added this metric to debug a specific problem in production (see where `ready_to_pop` got cancelled).

**Review by Dav1dde — APPROVED** [review]
Nice.

**Pull request merged**